aten/src/ATen/native/Math.h

```diff
@@ -1291,3 +1291,500 @@ calc_i0e(T _x) {
+}
+case 93: {
+T t = 2*y100 - 187;
+return 0.74993321911726254661e0 + (0.16272790364044783382e-1 + (0.20195505163377912645e-3 + (0.18269894883203346953e-5 + (0.12335161021630225535e-7 + (0.61523068312169087227e-10 + 0.21395783431111111111e-12 * t) * t) * t) * t) * t) * t;
+}
+case 94: {
+T t = 2*y100 - 189;
+return 0.78330143531283492729e0 + (0.17102934132652429240e-1 + (0.21321800585063327041e-3 + (0.19281661395543913713e-5 + (0.12963340087354341574e-7 + (0.64126040998066348872e-10 + 0.21986708942222222222e-12 * t) * t) * t) * t) * t) * t;
+}
+case 95: {
+T t = 2*y100 - 191;
+return 0.81837581041023811832e0 + (0.17979364149044223802e-1 + (0.22510330592753129006e-3 + (0.20344732868018175389e-5 + (0.13617902941839949718e-7 + (0.66799760083972474642e-10 + 0.22574701262222222222e-12 * t) * t) * t) * t) * t) * t;
+}
+case 96: {
+T t = 2*y100 - 193;
+return 0.85525144775685126237e0 + (0.18904632212547561026e-1 + (0.23764237370371255638e-3 + (0.21461248251306387979e-5 + (0.14299555071870523786e-7 + (0.69543803864694171934e-10 + 0.23158593688888888889e-12 * t) * t) * t) * t) * t) * t;
+}
+case 97: {
+T t = 2*y100 - 195;
+return 0.89402868170849933734e0 + (0.19881418399127202569e-1 + (0.25086793128395995798e-3 + (0.22633402747585233180e-5 + (0.15008997042116532283e-7 + (0.72357609075043941261e-10 + 0.23737194737777777778e-12 * t) * t) * t) * t) * t) * t;
+}
+case 98: {
+T t = 2*y100 - 197;
+return 0.93481333942870796363e0 + (0.20912536329780368893e-1 + (0.26481403465998477969e-3 + (0.23863447359754921676e-5 + (0.15746923065472184451e-7 + (0.75240468141720143653e-10 + 0.24309291271111111111e-12 * t) * t) * t) * t) * t) * t;
+}
+case 99: {
+T t = 2*y100 - 199;
+return 0.97771701335885035464e0 + (0.22000938572830479551e-1 + (0.27951610702682383001e-3 + (0.25153688325245314530e-5 + (0.16514019547822821453e-7 + (0.78191526829368231251e-10 + 0.24873652355555555556e-12 * t) * t) * t) * t) * t) * t;
+}
+  }
+  // we only get here if y = 1, i.e. |x| < 4*eps, in which case
+  // erfcx is within 1e-15 of 1..
+  return 1.0;
+}
+
+template <typename T>
+static inline typename std::enable_if<std::is_floating_point<T>::value, T>::type
+calc_erfcx(T x)
+{
+  if (x >= 0) {
+    if (x > 50) { // continued-fraction expansion is faster
+      const T ispi = 0.56418958354775628694807945156; // 1 / sqrt(pi)
+      if (x > 5e7) { // 1-term expansion, important to avoid overflow
+        return ispi / x;
+      }
+      /* 5-term expansion (rely on compiler for CSE), simplified from:
+                ispi / (x+0.5/(x+1/(x+1.5/(x+2/x))))  */
+      return ispi*((x*x) * (x*x+4.5) + 2) / (x * ((x*x) * (x*x+5) + 3.75));
+    }
+    return erfcx_y100(400/(4+x));
+  }
+  else {
+    return x < -26.7 ? std::numeric_limits<double>::max() : (x < -6.1 ? 2*exp(x*x)
```

Comment:
`SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /var/lib/jenkins/workspace/aten/src/ATen/native/Math.h:1787:12 in`
Guess need to rework this condition separately for float32.

Reply:
I think `std::numeric_limits<T>` should do the trick.

Reply:
Using `std::numeric_limits<>::max()` may introduce some strange behavior as the cutoff -26.7 does not capture the actual range for inf (See 2. and 3. test results below).

On the other hand `std::numeric_limits<T>::infinity()` behaves consistently with scipy.

```
import torch
from scipy import special  
# from torch import special

x_float32 = torch.linspace(-30, 30, 6001, dtype=torch.float32)
x_float64 = torch.linspace(-30, 30, 6001, dtype=torch.float64)

y_float32 = special.erfcx(x_float32)
y_float64 = special.erfcx(x_float64)

x_inf_float32 = x_float32[ y_float32 == float('Inf') ]
x_inf_float64 = x_float64[ y_float64 == float('Inf') ]

print( f'Range of inf for float32: [{x_inf_float32[0]}, {x_inf_float32[-1]}]' )
print( f'Range of inf for float64: [{x_inf_float64[0]}, {x_inf_float64[-1]}]' )
```
1) Use scipy.special
```
Range of inf for float32: [-30.0, -9.390000343322754]
Range of inf for float64: [-30.0, -26.63]
```
2) Use torch.special with `std::numeric_limits<double>::max()`
```
Range of inf for float32: [-30.0, -9.390000343322754]
Range of inf for float64: [-26.7, -26.63]
```
3) Use torch.special with `std::numeric_limits<T>::max()`
```
Range of inf for float32: [-26.69999885559082, -9.390000343322754]
Range of inf for float64: [-26.7, -26.63]
```
4) Use torch.special with `std::numeric_limits<T>::infinity()`
```
Range of inf for float32: [-30.0, -9.390000343322754]
Range of inf for float64: [-30.0, -26.63]
```